google/datalab/contrib/bigquery/operators/_bq_execute_operator.py

```diff
@@ -31,13 +30,15 @@ def __init__(self, sql, parameters=None, table=None, mode=None, *args, **kwargs)
 
   def execute(self, context):
     query = bq.Query(sql=self._sql)
-    # allow_large_results is set to False because setting it True requires the destination table to
-    # be specified.
-    # use_cache is set to False since this is most likely the case for all pipeline scenarios
+    # use_cache is False since this is most likely the case in pipeline scenarios
+    # allow_large_results can be True only if table is specified (i.e. when it's not None)
     output_options = bq.QueryOutput.table(name=self._table, mode=self._mode, use_cache=False,
-                                          allow_large_results=False)
-    pydatalab_context = google.datalab.Context.default()
+                                          allow_large_results=self._table is not None)
+
     query_params = Pipeline._get_query_parameters(self._parameters)
-    # TODO(rajivpb): Is this a sync or an async operation? Unlike load(), this does not wrap its
-    # async counterpart with a job.wait(). Test and wait if necessary.
-    query.execute(output_options, context=pydatalab_context, query_params=query_params)
+    job = query.execute(output_options, context=None, query_params=query_params)
```

Comment:
context should have a default; so no need to explicitly pass None.

To be honest, I am not sure why parameter has been added to all these methods like `execute`. I would have expected these to only have been in the ctors for various objects - Query, Table, etc. so as to provide the project/credentials and any other config to use when making API calls pertaining to the object in question.

Reply:
Removed context now.

Agreed, I wondered about that too. Perhaps we can sweep it up in refactoring-fixit.